signals.go

```diff
@@ -18,13 +17,7 @@ const signalBufferSize = 2048
 // while still forwarding all other signals to the process.
 // If notifySocket is present, use it to read systemd notifications from the container and
 // forward them to notifySocketHost.
-func newSignalHandler(enableSubreaper bool, notifySocket *notifySocket) *signalHandler {
-	if enableSubreaper {
-		// set us as the subreaper before registering the signal handler for the container
-		if err := system.SetSubreaper(1); err != nil {
-			logrus.Warn(err)
-		}
-	}
+func newSignalHandler(notifySocket *notifySocket) *signalHandler {
```

Comment:
As @kolyshkin pointed out in https://github.com/opencontainers/runc/pull/4278#issuecomment-2118530548
This PR has done such things. But there is still one thing I can't determine whether we should do or not:

If we use `PR_SET_CHILD_SUBREAPER` and `fork(2)` to replace `clone(CLONE_PARENT)`, I think we should move this signal handler to `libcontainer`. Or else someone use `libcontainer` directly in the code will have to write a signal handler by themselves.